### PR TITLE
cortex-m85: Add MPU PXN example

### PR DESCRIPTION
**Pull request by AhmedIsmail02**
Description
-----------

A new MPU region attribute **Privileged eXecute Never (PXN)** is introduced in Armv8.1-M architecture, where if an MPU region has PXN attribute set and the processor attempts to execute the code inside with privileged level, the Memory Management Fault exception would be triggered, with **IACCVIOL** bit in MemManage Fault State Register set to 1. The PXN feature allows privileged software to ensure specific application tasks (threads) to execute in unprivileged level only. For example, a hacker cannot use stack corruption in a privileged peripheral handler to branch into unprivileged codes and execute them with privileged level.
Please refer to the Security related enhancements section in Introduction to Armv8.1-M architecture white paper document for more information.

This example demonstrates how the new MPU region attribute **Privileged eXecute Never (PXN)** introduced in Armv8.1-M architecture can be used on Cortex-M85 MCU. The example is based on **Corstone-310** Ecosystem Fixed Virtual Platform (Arm Cortex-M85 CPU and Ethos-U55 NPU).

The example simulates a privileged task trying to execute an instruction from an MPU region with **Privileged eXecute Never (PXN)** attribute set, this result in a memory fault exception which recovers gracefully after clearing an entry in a fault tracker buffer indicating that the exception was triggered and recovered.

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by kstribrnAmzn — APPROVED** [review]

**Comment by AhmedIsmail02** [comment]
Hello @kstribrnAmzn ,
Thanks for the review, are there any blockers for this PR to get merged?

**Comment by kstribrnAmzn** [comment]
None at all. I will merge it now! I thought once I'd approved you'd be able to merge.

**Comment by AhmedIsmail02** [comment]
@kstribrnAmzn I'm not sure if i have the merging rights, thank you for your support!

**Pull request merged**
